src/test/macromanager_test.cpp

```diff
@@ -0,0 +1,85 @@
+#include "recording/macromanager.h"
+
+#include <gtest/gtest.h>
+
+#include <QDebug>
+
+#include "mixxxtest.h"
+#include "signalpathtest.h"
+
+TEST(MacrosTest, CreateMacro) {
+    auto macro = new Macro();
+    ASSERT_EQ(macro->m_length, 0);
+}
+
+TEST(MacroManagerTest, ClaimRecording) {
+    auto mgr = MacroManager();
+    EXPECT_EQ(mgr.isRecordingActive(), false);
+    mgr.claimRecording();
+    EXPECT_EQ(mgr.isRecordingActive(), false);
+    mgr.m_macroRecordingState.store(MacroState::Armed);
+    mgr.claimRecording();
+    EXPECT_EQ(mgr.isRecordingActive(), true);
+}
+
+TEST(MacroManagerTest, RecordCueJump) {
+    auto mgr = MacroManager();
+    ChannelHandle handle = ChannelHandleFactory().getOrCreateHandle("");
+    EXPECT_EQ(mgr.m_macroRecordingState.load(), MacroState::Disabled);
+    mgr.notifyCueJump(handle, 0, 1);
+    EXPECT_EQ(mgr.m_activeChannel, nullptr);
+    EXPECT_EQ(mgr.getMacro().m_length, 0);
+    mgr.m_macroRecordingState.store(MacroState::Armed);
```

Comment:
These tests are too tightly coupled with MacroManager internals and will be hard to maintain long term. I want tests to fail I made a mistake, not break compilation each time I change the name of a private member variable.
```suggestion
    EXPECT_EQ(mgr.getRecordingState(), MacroState::Disabled);
    mgr.notifyCueJump(handle, 0, 1);
    EXPECT_EQ(mgr.getActiveChannel(), nullptr);
    EXPECT_EQ(mgr.getMacro().getLength(), 0);
    mgr.setRecordingState(MacroState::Armed);
```

Reply:
The problem is that I don't want to add superfluous methods just for the test. Some of the suggested things do make sense though. I'll see.

But if you rename the variable, don't you usually also rename the getters? And does your IDE not support proper refactoring, which automatically renames all occurences? ;)